core/src/main/java/nl/nn/adapterframework/validation/XSD.java

```diff
@@ -261,17 +264,30 @@ public int compareToByContents(XSD x) {
 	@Override
 	public abstract Reader getReader() throws IOException;
 
+	public static Set<IXSD> getXsdsRecursive(Set<IXSD> xsds) throws ConfigurationException {
+		return getXsdsRecursive(xsds, false);
+	}
 
-	@Override
-	public Set<IXSD> getXsdsRecursive(boolean supportRedifine) throws ConfigurationException {
-		return getXsdsRecursive(new HashSet<IXSD>(), supportRedifine);
+	public static Set<IXSD> getXsdsRecursive(Set<IXSD> xsds, boolean supportRedefine) throws ConfigurationException {
+		Map<String,IXSD> xsdsRecursive = new LinkedHashMap<>();
+		int i=0;
+		for (IXSD xsd : xsds) {
+			// All top level XSDs need to be added, with a unique systemId. If they come from a WSDL, they all have the same systemId.
+			String normalizedSystemId = FilenameUtils.normalize(xsd.getSystemId())+"["+(++i)+"]";
+			xsdsRecursive.put(normalizedSystemId, xsd);
+			xsd.getXsdsRecursive(xsdsRecursive, supportRedefine);
+		}
+		Set<IXSD> allXsds = new LinkedHashSet<>();
+		allXsds.addAll(xsdsRecursive.values());
+		return allXsds;
```

Comment:
Waarom doe je dit?

Reply:
De oorsponkelijke getXsdsRecursive()s werkten allemaal met een Set, maar daarmee kan je niet goed duplicate XSDs vinden. De nieuwe werkt met een Map, met als key systemId. Ik wilde nu niet het interface voor de aanroepende partij veranderen, daarom wordt de inhoud van de Map hier weer omgezet naar een Set.
Ik kan me voorstellen dat het later ook beter is op andere plekken ook de Map te gaan gebruiken.